### PR TITLE
v1.13 backports 2023-02-12

### PR DESCRIPTION
**Pull request by pchaigno**
- [x] #22657 -- Revert "images/cilium-test: New test suite image" (@pchaigno)
     - Bunch of trivial conflicts.
 - [x] #23202 -- Introduce node IDs (@pchaigno)
     - A lot of minor conflicts.
 - [x] #23408 -- clustermesh docs: add global and shared services reference (@giorio94)
 - [x] #23466 -- clustermesh: close etcd connection on config retrieval error (@giorio94)
 - [ ] #23429 -- fqdn/dnsproxy: move init LRU cache call out of StartDNSProxy. (@tommyp1ckles)
 - [x] #23467 -- etcd kvstore: rate limit watch retries on list errors (@giorio94)
 - [x] #23395 -- Uniform the style of the annotations used in the Cilium codebase (@giorio94)
 - [x] #23505 -- test: Remove IPsec + ep routes + VXLAN from ginkgo tests (@pchaigno)
 - [ ] #23404 -- docs: correct Prometheus port (@lizrice)
 - [ ] #23304 -- docs: refactor AKS installation instructions (@nbusseneau)
 - [x] #23393 -- docs,daemon: Added discouragement warnings for MetalLB to docs and agent (@dylandreimerink)
 - [x] #23100 -- docs: Add guide for proxy load balancing feature (@sayboras)
 - [ ] #23529 -- egressgw: cache matching endpoints in policy config (@jibi)
 - [x] #23534 -- gha: Rename ConformanceKind1.19 to ConformanceKind (@sayboras)
 - [x] #22797 -- bugtool: Dump envoy metrics for troubleshooting (@sayboras)
 - [x] #23118 -- docs: Drop sphinxcontrib-openapi fork, switch back to upstream (@qmonnet)
 - [x] #23516 -- Uniform leftover annotations in clustermesh docs (@giorio94)
 - [ ] #23574 -- examples, docs: simplify manifest use in clustermesh global services example (@tklauser)
 - [x] #23607 -- hubble/metrics: handle SCTP port in flows to world metrics (@rolinh)
 - [x] #23598 -- ci: Update docs-builder image for documentation workflow (@qmonnet)
 - [ ] #23555 -- daemon: Deprecate SockOps (@brb)
 - [x] #23546 -- docs: Link KNP sections together to reduce duplication (@christarazi)
 - [x] #23506 -- cilium-health status: fix endpoint reachability in succinct view (@giorio94)
 - [ ] #23578 -- Node ID restoration (@pchaigno)
 - [x] #23553 -- ipsec + ipv6-only nodes: add limitation note and better error handling (@giorio94)
 - [ ] #23588 -- Misc. clustermesh debug cleanups (@tklauser)
 - [ ] #23642 -- fix:file fd should be closed (@jiuker)

Skipped:

 - [ ] #23292 -- resource: Work around a rare race in initial sync (@joamaki)
     - I wasn't confident enough to backport this given conflicts.
 - [ ] #23399 -- bpf/test: Get rid of 4.9 leftovers (@brb)
     - Verifier tests started to fail. Changed the label of that PR to `backport/author`

Once this PR is merged, you can update the PR labels via:
```upstream-prs
$ for pr in 22657 23202 23408 23466 23429 23467 23395 23505 23404 23304 23393 23100 23529 23534 22797 23118 23516 23574 23607 23598 23555 23546 23506 23578 23553 23588 23642; do contrib/backporting/set-labels.py $pr done 1.13; done
```

### CODE REVIEW

**Comment by pchaigno** [comment]
/test-backport-1.13

**Review by sayboras — APPROVED** [review]
Thans and looks good for my commits :+1:

**Review by dylandreimerink — APPROVED** [review]

**Review by christarazi — APPROVED** [review]

**Review by giorio94 — APPROVED** [review]
Thanks!

**Review by kaworu — APPROVED** [review]
Thanks @pchaigno, Hubble changes LGTM.

File: test/k8s/verifier.go [hunk 1]
```diff
@@ -93,7 +93,7 @@ var _ = Describe("K8sDatapathVerifier", func() {
 	}
 
 	getKernel := func() string {
-		kernel := "49"
```
Comment (julianwiedmann) [inline test/k8s/verifier.go]: We're still running the verifier job on 4.9 (as https://github.com/cilium/cilium/pull/22754/commits/393e1004c4af51346fc51f7bfb5250b4ba108d8b hasn't been backported yet).

But this change means that it now uses the 5.4 complexity configs. Note how eg https://jenkins.cilium.io/job/Cilium-PR-K8s-1.19-kernel-4.9/2628/testReport/junit/Suite-k8s-1/19/K8sDatapathVerifier_Runs_the_kernel_verifier_against_Cilium_s_BPF_datapath/ contains `-DENABLE_EGRESS_GATEWAY=1` - that doesn't exist in a 4.9 config.

**Comment by aanm** [comment]
/test-backport-1.13

**Review by rolinh — APPROVED** [review]
Thanks looks good for my commit.

**Review by qmonnet — APPROVED** [review]

**Comment by aanm** [comment]
/test-1.24-4.9

**Comment by aanm** [comment]
/test-1.20-4.9

**Review by nathanjsweet — APPROVED** [review]
lgtm for api changes.

**Pull request merged**